generics: C error when there's an existing function name

<!-- You can use `v doctor` to fill up the next fields -->
**OS:** linux, Ubuntu 20.04 LTS
**Processor:** 8 cpus, 64bit, little endian, Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz**
**CC version:** cc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
**V full version:** V 0.1.29 9fbf38c

**What did you do?**
Tried to test if a type-specific function would be possible in conjunction with a similarly-named generic function.
```v
module main

fn get<T>(typ T) T {
   return typ
}

fn get_string(typ string) string {
   return 'boom'
}

fn main() {
   println(get<string>('hello'))
}
```

**What did you expect to see?**
~~V compiler error~~ Successful compilation with the `get_string` function being renamed to distinguish to the generic one
 
**What did you see instead?**
Redefinition error in C
```
/tmp/v/gen_t.3337294832360845231.tmp.c:8058:15: error: redefinition of ‘main__get_string’
 8058 | static string main__get_string(string typ) {
      |               ^~~~~~~~~~~~~~~~
/tmp/v/gen_t.3337294832360845231.tmp.c:8054:15: note: previous definition of ‘main__get_string’ was here
 8054 | static string main__get_string(string typ) {
      |               ^~~~~~~~~~~~~~~~
...

--- a/vlib/v/gen/fn.v
+++ b/vlib/v/gen/fn.v
@@ -60,9 +60,10 @@ fn (mut g Gen) gen_fn_decl(it ast.FnDecl, skip bool) {
 	}
 	mut type_name := g.typ(it.return_type)
 	if g.cur_generic_type != 0 {
-		// foo<T>() => foo_int(), foo_string() etc
+		// foo<T>() => foo_T_int(), foo_T_string() etc
 		gen_name := g.typ(g.cur_generic_type)
-		name += '_' + gen_name
+		// Using _T_ to differentiate betweem get<string> nad get_string
+		name += '_T_' + gen_name
 	}
 	// if g.pref.show_cc && it.is_builtin {
 	// println(name)
@@ -534,8 +535,9 @@ fn (mut g Gen) fn_call(node ast.CallExpr) {
 		name = c_name(name)
 	}
 	if node.generic_type != table.void_type && node.generic_type != 0 {
-		// `foo<int>()` => `foo_int()`
-		name += '_' + g.typ(node.generic_type)
+		// Using _T_ to differentiate betweem get<string> nad get_string
+		// `foo<int>()` => `foo_T_int()`
+		name += '_T_' + g.typ(node.generic_type)
 	}
 	// TODO2
 	// cgen shouldn't modify ast nodes, this should be moved

--- a/vlib/v/gen/fn.v
+++ b/vlib/v/gen/fn.v
@@ -62,7 +62,7 @@ fn (mut g Gen) gen_fn_decl(it ast.FnDecl, skip bool) {
 	if g.cur_generic_type != 0 {
 		// foo<T>() => foo_T_int(), foo_T_string() etc
 		gen_name := g.typ(g.cur_generic_type)
-		// Using _T_ to differentiate betweem get<string> nad get_string
+		// Using _T_ to differentiate between get<string> and get_string
 		name += '_T_' + gen_name
 	}
 	// if g.pref.show_cc && it.is_builtin {
@@ -535,7 +535,7 @@ fn (mut g Gen) fn_call(node ast.CallExpr) {
 		name = c_name(name)
 	}
 	if node.generic_type != table.void_type && node.generic_type != 0 {
-		// Using _T_ to differentiate betweem get<string> nad get_string
+		// Using _T_ to differentiate between get<string> and get_string
 		// `foo<int>()` => `foo_T_int()`
 		name += '_T_' + g.typ(node.generic_type)
 	}

--- a/vlib/v/tests/generic_functions_with_normal_function_test.v
+++ b/vlib/v/tests/generic_functions_with_normal_function_test.v
@@ -1,0 +1,12 @@
+fn get<T>(typ T) T {
+   return typ
+}
+
+fn get_string(typ string) string {
+   return 'boom'
+}
+
+fn test_generic_with_same_type() {
+	assert get_string('') == 'boom'
+	assert get<string>('hello') == 'hello'
+}

